### ISSUE TITLE
Combining `///`-style comments with macros containing @cond/@endcond causes a preprocessor error

### ISSUE THREAD

**Issue by chrisdembia** [doxygen/doxygen#7873]
Thanks @doxygen for such a great tool! I have used Doxygen for years now to document properties that are similar to `Q_PROPERTY`. The property is created by my macro, `DECLARE_PROPERTY`, and the macro generates doxygen documentation. This works fine unless the class using my macro uses `///`-style comments.

**Describe the bug**

Combining `///`-style comments with a macro that contains `@cond`/`@endcond` causes the preprocessor to fail.

I use the following input files:

doxyfile
```
# Doxyfile 1.8.18
QUIET                  = YES
INPUT                  = Bounds.h
ENABLE_PREPROCESSING   = YES
MACRO_EXPANSION        = YES
EXPAND_ONLY_PREDEF     = YES
EXPAND_AS_DEFINED      = DECLARE_PROPERTY
```

Bounds.h
```cpp
#ifndef BOUNDS_H
#define BOUNDS_H

#define DECLARE_PROPERTY(pname, T, comment)                                 \
    /** @cond */                                                            \
    /** Internal function for <b> pname </b> property.                   */ \
    void hidden_function_##pname();                                         \
    /** @endcond */                                                         \
    /** @name Properties (single-value)                                  */ \
    /** @{                                                               */ \
    /** comment                                                          */ \
    Q_PROPERTY(T, pname)                                                    \
    /** @}                                                               */ \
    /** @name Property-related functions                                 */ \
    /** @{                                                               */ \
    /** Get the value of the <b> pname </b> property.                    */ \
    const T& get_##pname() const;                                           \
    /** @}                                                               */

/** This works fine. */
class BoundsGood {
public:
    /** Constructor.
    Line two. */
    BoundsGood();
    DECLARE_PROPERTY(lower_bound, double, "Value of lower bound.");
    DECLARE_PROPERTY(upper_bound, double, "Value of upper bound.");
};

/** This causes a warning and the class ends up not documented. */
class BoundsBad {
public:
    /// Constructor.
    /// Line two.
    BoundsBad();
    DECLARE_PROPERTY(lower_bound, double, "Value of lower bound.");
    DECLARE_PROPERTY(upper_bound, double, "Value of upper bound.");
};

#endif
```

Running `doxygen doxyfile` with doxygen 1.8.18 generates the following output:
```
/Users/chris/Documents/doxygen/macrobug/Bounds.h:37: warning: Conditional section does not have a corresponding \endcond command within this file.
/Users/chris/Documents/doxygen/macrobug/Bounds.h:36: warning: Conditional section does not have a corresponding \endcond command within this file.
/Users/chris/Documents/doxygen/macrobug/Bounds.h:41: warning: Reached end of file while still inside a (nested) comment. Nesting level 1 (probable line reference: 37)
*/   /** Internal function for <b>  lower_bound  </b> property.                   */   void hidden_function_lower_bound ();   /** @endcond */   /** @name Properties (single-value)                                  */   /** @{                                                               */   /**  "Value of lower bound."                                                           */   Q_PROPERTY( double ,  lower_bound )   /** @}                                                               */   /** @name Property-related functions                                 */   /** @{                                                               */   /** Get the value of the <b>  lower_bound  </b> property.                    */   const  double & get_lower_bound () const;   /** @}                                                               */;*/   /** Internal function for <b>  upper_bound  </b> property.                   */   void hidden_function_upper_bound ();   /** @endcond */   /** @name Properties (single-value)                                  */   /** @{                                                               */   /**  "Value of upper bound."                                                           */   Q_PROPERTY( double ,  upper_bound )   /** @}                                                               */   /** @name Property-related functions                                 */   /** @{                                                               */   /** Get the value of the <b>  upper_bound  </b> property.                    */   const  double & get_upper_bound () const;   /** @}                                                               */;
```

Not only is a warning generated, but `doxygen` emits a portion of the macro. Both the warning and the emitted macro are unexpected.

**Expected behavior**

No warnings should be generated and the generated documentation for `BoundsGood` and `BoundsBad` should be the same.

**Screenshots**
If useful, add screenshots to help explain your problem.

This is the html output for `BoundsGood`:
<img width="318" alt="image" src="https://user-images.githubusercontent.com/846001/85788054-361d2d00-b6e1-11ea-84a9-a754895fdf08.png">

No output is generated for class `BoundsBad`.

**To Reproduce**
[doxygen_macro_cond_bug.zip](https://github.com/doxygen/doxygen/files/4833463/doxygen_macro_cond_bug.zip)

**Version**
doxygen 1.8.18 on macOS Catalina 10.15.5, installed from from Homebrew

**Comment by albert-github** [comment]
Strange behavior for the `///` version. I also get:
```
*/   /** Internal function for <b>  lower_bound  </b> property.                   */   void hidden_function_lower_bound ();   /** @endcond */   /** @name Properties (single-value)                                  */   /** @{                                                               */   /**  "Value of lower bound."                                                           */   Q_PROPERTY( double ,  lower_bound )   /** @}                                                               */   /** @name Property-related functions                                 */   /** @{                                                               */   /** Get the value of the <b>  lower_bound  </b> property.                    */   const  double & get_lower_bound () const;   /** @}                                                               */;*/   /** Internal function for <b>  upper_bound  </b> property.                   */   void hidden_function_upper_bound ();   /** @endcond */   /** @name Properties (single-value)                                  */   /** @{                                                               */   /**  "Value of upper bound."                                                           */   Q_PROPERTY( double ,  upper_bound )   /** @}                                                               */   /** @name Property-related functions                                 */   /** @{                                                               */   /** Get the value of the <b>  upper_bound  </b> property.                    */   const  double & get_upper_bound () const;   /** @}                                                               */;
```
on the output.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #7875 for the extra output as shown in https://github.com/doxygen/doxygen/issues/7873#issuecomment-650100405 (not the root cause of this issue).

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #7876 for the real problem.

**Comment by albert-github** [comment]
The code of the secondary problem (extra output #7875) has been integrated in master on github

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.4.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
